plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/build/XStatefulIncrementalBuilder.java

```diff
@@ -103,34 +103,52 @@ public XBuildResult launch() {
 				removeGeneratedFiles(source, newSource2GeneratedMapping);
 			}
 
-			XIndexer.XIndexResult result = indexer.computeAndIndexAffected(request, context);
+			List<Delta> allProcessedAndExternalDeltas = new ArrayList<>(request.getExternalDeltas());
+
+			ResourceDescriptionsData oldIndex = context.getOldState().getResourceDescriptions();
+			Set<URI> remainingURIs = new LinkedHashSet<>(oldIndex.getAllURIs()); // note: creating a copy!
```

Comment:
No need for `LinkedHashSet`

Reply:
Do you mean no need for a "linked" set or no need for a set at all?

Reply:
In any case, I think we need
* a set because many URIs are removed from this collection over time (see line 132),
* a "linked" set, because the implementation of `#computeAndIndexAffected()` is iterating over `remainingURIs` and indirectly defining the build order during the next build cycle.

Reply:
My thinking was that the `LinkedHashSet` can be replaced by a plain `HashSet`

Reply:
+1 to keep it as is.